Collector/ProfileFactory.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+class ProfileFactory
```

Comment:
as there are no interfaces on profile and factory: why not just do `new` in the place where we need a profile? as these classes are internal anyways, they do not need to be extensible. having factory and all this invites people to customize...